### ISSUE TITLE
Replace the rich-text-editor

### ISSUE THREAD

**Issue by chandanbn** [Vulnogram/Vulnogram#57]
The current rich-text editor [wysihtml5](http://wysihtml.com/), hasn't been updated since 6 years and has some known issues. It is about 500kb and throws errors on the javascript console.

A replacement should meet these requirements:
1. Only allow some predefined list of tags (like a, b, i, table, p, img ...) and some essential attributes like href.
2. When rich-text content is pasted, or drag-dropped, it should only preserve the allowed tags and attributes, and drop fonts, colors, sizes etc., 
3. should be small and fast loading
4. Should be able to set or get html value
5. Free or open source license
6. Allowing simultaneous collaborative editing would be a plus

Potential candidates to consider 
https://tiptap.dev/ 
https://github.com/microsoft/roosterjs

**Comment by sei-vsarvepalli** [comment]
Not considering tinymce? https://www.npmjs.com/package/tinymce - looks active project.  

I assume no.1 in your list is about security, one may need to extend or escape content for clean backend storage.

**Comment by chandanbn** [comment]
Yes, looking for something minimal but functional that should not load anything other than some allowed list of text formatting tags (either from backend or from copy/paste, drag drop sources)

**Comment by jzebor** [comment]
Chrome browser console logs a message about 'DOMNodeRemoved' mutation event being deprecated. According to the [announcement](https://chromestatus.com/feature/5083947249172480 ), Chrome will stop supporting this in July 2024. Do you know how/if this will impact Vulnogram ?
